### PR TITLE
Validation and documentation

### PR DESCRIPTION
**Pull request by n-a-t-e**
This is a pretty big one and more sprawling than usual. Here's what happened:

 - added logic to validate records - date validation, missing bilingual fields, missing EOVs in keywords, etc
 - moved all validation code to it's own file for clarity
 - added many tests
 - added the code to `escape()` text and removed fields in record before becoming XML
 - reduced the number of required fields for now
 - made sample french and english minimum YAML records
 - status defaults to  onGoing
 - date defaults to now - the schema requires at least one date
 
 - Expanded and clarified the README
 - changed how the module is run from the command line to use `python -m metadata_xml` so that it can be more easily run from any directory/cron script

### CODE REVIEW

File: metadata_xml/cioos_template.jinja2 [hunk 1]
```diff
@@ -96,19 +96,17 @@
...
   {% endif %}
 
-  {% if record['scope_code'] %}
   {# metadataScope: mandatory #}
   <mdb:metadataScope>
     <mdb:MD_MetadataScope>
       {# resourceScope: mandatory #}
       <mdb:resourceScope>
-        <mcc:MD_ScopeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_ScopeCode" codeListValue="{{ record['scope_code'] }}"/>
+        <mcc:MD_ScopeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xmlMD_ScopeCode_dataset" codeListValue="dataset"/>
```
Comment (jeffcullis) [inline metadata_xml/cioos_template.jinja2]: Interesting that the '#' isn't necessary here, didn't realize that

**Comment by jeffcullis** [comment]
Thanks Nate, this all looks good to me. Nice work with the minimal yaml validation and tests.

File: metadata_xml/cioos_template.jinja2 (hunk 1 above)
Comment (n-a-t-e) [reply metadata_xml/cioos_template.jinja2]: oh that is a typo, will fix

File: metadata_xml/cioos_template.jinja2 [hunk 2]
```diff
@@ -101,7 +101,7 @@
     <mdb:MD_MetadataScope>
       {# resourceScope: mandatory #}
       <mdb:resourceScope>
-        <mcc:MD_ScopeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xmlMD_ScopeCode_dataset" codeListValue="dataset"/>
+        <mcc:MD_ScopeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_ScopeCode" codeListValue="dataset"/>
```
Comment (jeffcullis) [inline metadata_xml/cioos_template.jinja2:104]: OK so the # did need to be there.

File: metadata_xml/record.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ title_eng: "title in english"
 summary: "summary in french"
 summary_eng: "summary in english"
 keywords: "kw1 in french,kw2 in french"
-keywords_eng: "kw1 in english,kw2 in english,kw3 in english"
+keywords_eng: "kw1 in english,kw2 in english,kw3 in english, oxygen"
```
Comment (jeffcullis) [inline metadata_xml/record.yaml:10]: Good that you've pointed out the keyword list is for EOVs here.

**Review by jeffcullis — APPROVED** [review]
Everything looks good. Thanks Nate!

**Pull request merged**